Add HTTP proxy policy setting for Template Sync (#3409)
theforeman/foreman-documentation: guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc

Use this procedure to synchronize templates between your {ProjectServer} and an existing repository.

.Procedure
. Configure connection to the repository:

> .. To use an HTTP proxy to connect to the repository, ensure the proxy is configured on your {ProjectServer}.
> In the {ProjectwebUI}, navigate to *Infrastructure > HTTP Proxies*.

Reviewer: ```suggestion
```
I wouldn't mention this here, choosing proxy fundamentally doesn't differ from choosing repo or branch.
On the other hand, it would probably be worth mentioning that this entire step can be skipped if the user wants to connect to the repository using an HTTP proxy, an HTTPS proxy, or a regular HTTP connection.

Author: Ok, I took another look at the whole procedure and tried to streamline it a bit.

----
... Allow the `foreman` user access to the `.config` directory:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
# chown --recursive foreman /usr/share/foreman/.config
----
... Update the Git global configuration for the `foreman` user with the path to your self-signed CA certificate:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
# sudo --user foreman git config --global http.sslCAPath _Path_To_CA_Certificate_
----
.. To use SSH to connect to the repository:
... Create an SSH key pair if you do not already have it.
Do not specify a passphrase.
+
----
# sudo --user foreman ssh-keygen
----
... Configure your version control server with the public key from your {Project}, which resides in `/usr/share/foreman/.ssh/id_rsa.pub`.
... Accept the Git SSH host key as the `foreman` user:
+
[subs="+quotes"]
----
# sudo --user foreman ssh _git.example.com_
----
. In the {ProjectWebUI}, navigate to *Administer > Settings*.
On the *Template Sync* tab, provide details for your Git repository.
.. Change the *Branch* setting to match the target branch on a Git server.
.. Change the *Repo* setting to match the Git repository.

> For example, for the repository located in `git@git.example.com/templates.git` set the setting into `git@git.example.com/templates.git`.

Author: Nice! Applied, somewhat modified.

----
# sudo --user foreman ssh _git.example.com_
----
. In the {ProjectWebUI}, provide details for your Git repository:

> .. Navigate to *Administer > Settings*.

Reviewer 2: ```suggestion
.. Navigate to *Administer* > *Settings*.
```

.. Click the *Template Sync* tab.
.. Change the *Branch* setting to define the target branch of the repository.
.. Change the *Repo* setting to define the repository URL.

> .. If you want to use an HTTP proxy to connect to the repository, change the *HTTP proxy policy* setting to define your HTTP proxy policy.

Reviewer: Maybe something like this would add a bit more clarity? Although I am not certain if I am not being overly specific.
```suggestion
.. If you want to use an HTTP proxy to connect to the repository, change the *HTTP proxy policy* setting to `Global default HTTP proxy` or `Custom HTTP proxy`.
If you select `Custom HTTP proxy`, you will be prompted to select the HTTP proxy when syncing templates.
```

Author: I think that considering what the web UI will say, we can go with only the first of the two lines you're suggesting. Thanks!

> .. If you want to use an HTTP proxy to connect to the repository, change the *HTTP proxy policy* setting to define your HTTP proxy policy.

Reviewer: There are many settings on the template sync tab that that affect the template sync (or rather eliminate the need to input them manually while syncing). Maybe you could mention that it is possible to configure these settings too as an optional step, possibly with the link to template sync settings. But again, this is more stylistic than technical suggestion so apply or disregard at your own discretion.

Author: At this point in the procedure, users will see the settings along with descriptions of what the settings do. I don't think that we need to explicitly tell them that those settings are there at a moment when they can see them for themselves. We do mention the settings in the 'next steps' procedures (importing/exporting templates) . These procedures actually refer back to the Template Sync tab. I think that's enough.

So, no change.

.Next steps

> 

Reviewer 2: ```suggestion
```

* To import templates from your repository, continue with xref:Importing_Templates_{context}[].

> * To export templates from your repository, continue with xref:Exporting_Templates_{context}[].

Reviewer: ```suggestion
* To export templates to your repository, continue with xref:Exporting_Templates_{context}[].
```
